### DISCUSSION TITLE
Cannot use layout class component with propeties

### DISCUSSION THREAD

**Discussion by ph7jack** [livewire/livewire#3493] [7. Bugs (archived)]
When i set my `AppLayout::class` in `livewire.layout` config, the variables is not initialized, the constructor is not called

I'm using the latest version of livewire

**Comment by ryatkins** [comment]
Where are you trying to do this?

Can you do it on the render method in your component and call the name of your layout via string?

```
    public function render()
    {
        return view('livewire.show-posts')
            ->layout('layouts.base');
    }
```

https://laravel-livewire.com/docs/2.x/rendering-components#custom-layout

**Comment by devcircus** [comment]
I believe you should be using a string in the config, using dot-notation according to the structure of your "views" directory, not a class.

Thanks,

Clay
________________________________

**Comment by joshhanley** [comment]
@PH7-Jack Livewire doesn't support class based layouts in the config file at the moment. There is currently an open PR to add support https://github.com/livewire/livewire/pull/3363
